traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -976,8 +981,13 @@ func getServers(h http.Header, params map[string]string, tx *sqlx.Tx, user *auth
 	}
 
 	returnable := make([]tc.ServerNullable, 0, len(ids))
+
 	for _, id := range ids {
 		server := servers[id]
+		// if requesting servers for a given ds, need to only include those in the same cdn
+		if len(dsCDNName) > 0 && string(dsCDNName) != *server.CDNName {
+			continue
+		}
```

Comment:
Is there a way to perform this check in the query itself?

Reply:
yeah, i think you're right. best if done in the query but i had luck figuring that out. i'll look again.

Reply:
@zrhoffman - updated. how's that look?